Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "PSTypes.h"
+#include "Subsystems/WorldSubsystem.h"
+#include "PSWorldSubsystem.generated.h"
+
+
+/**
+ * Implements the world subsystem to access different components in the module 
+ */
+UCLASS(BlueprintType, Blueprintable, Config = "ProgressionSystem", DefaultConfig)
+class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
+{
+	GENERATED_BODY()
+
+	DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FCurrentRowDataChanged, const FPSRowData&, SavedProgressionRowDataInternal);
+
+public:
+	/** Returns this Subsystem, is checked and wil crash if can't be obtained.*/
+	static UPSWorldSubsystem& Get();
+	static UPSWorldSubsystem& Get(const UObject& WorldContextObject);
+
+
+	/* Delegate for informing row data changed */
+	UPROPERTY(BlueprintAssignable, Category = "C++")
+	FCurrentRowDataChanged OnCurrentRowDataChanged;
+
+	/** Returns the data asset that contains all the assets of Progression System game feature.
+	 * @see UPSWorldSubsystem::PSDataAssetInternal. */
+	UFUNCTION(BlueprintCallable, BlueprintPure, Category = "C++")
+	const class UPSDataAsset* GetPSDataAsset() const;
+
+	/** Returns a progression System component reference */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE class UPSHUDComponent* GetProgressionSystemHUDComponent() const { return PSHUDComponentInternal; }
+
+	/** Returns a current row data */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	const FORCEINLINE FPSRowData GetCurrentRowData() { return SavedProgressionRowDataInternal; }
+
+	/** Returns a current save game instance */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE UPSSaveGameData* GetCurrentSaveGameData() const { return SaveGameInstanceInternal; }
+
+	/** Set the progression system component */
+	UFUNCTION(BlueprintCallable, Category = "C++")
+	void SetProgressionSystemComponent(UPSHUDComponent* MyProgressionSystemComponent);
+
+	/** Saves the progression to the local files */
+	UFUNCTION()
+	void SaveDataAsync();
+
+protected:
+	/** Contains all the assets and tweaks of Progression System game feature.
+	 * Note: Since Subsystem is code-only, there is config property set in BasePSDataAssetInternal.ini.
```

Comment:
In this comment unexisting filename is mentioned: `BasePSDataAssetInternal.ini`. Did you mean `BaseProgressionSystem.ini`? Please correct it here.

Reply:
thanks for noticing! set to correct one